common/src/main/java/io/netty/util/internal/InternalThreadLocalMap.java

```diff
@@ -182,6 +188,21 @@ public StringBuilder stringBuilder() {
         return cache;
     }
 
+    public <E> ArrayList<E> arrayList() {
+        return arrayList(DEFAULT_ARRAY_LIST_INITIAL_CAPACITY);
+    }
+
+    public <E> ArrayList<E> arrayList(int minCapacity) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The return type of this method should be an interface such as "List" rather than the implementation "ArrayList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1319)